internal/table/table.go

```diff
@@ -133,51 +140,120 @@ func rescaleColumns(colWidths []int, targetTotal int) {
 	colWidths[len(colWidths)-1] += targetTotal - mathutils.Total(colWidths...)
 }
 
+func getCroppedText(text string, maxLen int) []entry {
+	entries := make([]entry, 0)
+	pos := 0
+	matches := colorize.ColorRx.FindAllSubmatchIndex([]byte(text), -1)
+	runeText := []rune(text)
+
+	for pos < len(runeText) {
+		buffer := bytes.Buffer{}
+		count := 0
+		for count < maxLen {
+			// If we reach an index that we recognize (ie. the start of a tag)
+			// then we write the whole tag, otherwise write by rune
+			if end, match := matchIndex(pos, matches); match {
+				buffer.WriteString(string(runeText[pos:end]))
+				pos = end
+			} else {
+				if pos > len(runeText)-1 {
+					break
+				}
+
+				if runeText[pos] == linebreakRune {
+					pos++
+					break
+				}
+
+				buffer.WriteString(string(runeText[pos]))
+				pos++
+				count++
+			}
+		}
+		entries = append(entries, entry{buffer.String(), count})
+	}
+
+	// Clean up any entries that are just color tags
+	// This can happen when the end of a string or line
+	// break occurs right before a tag
+	for i, entry := range entries {
+		if entry.length != 0 {
+			continue
+		}
+
+		text := entries[i].line
+		if i >= 1 {
+			entries[i-1].line = entries[i-1].line + text
+		}
+		entries = entries[0:i]
+	}
+
+	return entries
+}
+
+func matchIndex(pos int, matches [][]int) (int, bool) {
+	for _, match := range matches {
+		start, stop := match[0], match[1]
+		if pos == start {
+			return stop, true
+		}
+	}
+	return -1, false
+}
```

Comment:
http://github.com/ActiveState/cli/blob/master/internal/sliceutils/sliceutils.go#L9-L9

Reply:
This function is doing a little more than the one you linked, specifically it's giving me the ending position of the tag if the index matches the start. I've removed this function and added the loop into `GetCroppedText`